woocommerce/woocommerce-gutenberg-products-block
# Migrate attribute-filter block to use new interactivity store API

## What

Partially fixes https://github.com/woocommerce/woocommerce-blocks/issues/11610

## Why

When the new interactivity store API was introduced a lot of the experimental blocks we've built, broke due to the store API not being backwards compatible.

This refactors the collection attribute filter to work with the new API, sometimes improving it where it makes sense for the new API (such as renaming actions that now fall under a namespace).

This also fixes a small bug, when the attribute filter is alone without other blocks using the checkbox list styles it would break, this does the same thing we did for the stock-filter by explicitly loading those styles.

I think we should turn the checkbox list into a component and have it load the styles itself but that can be done in future as a follow up.

## Testing Instructions

You can use the testing instructions from the [original MVP PR](https://github.com/woocommerce/woocommerce-blocks/pull/11749) to test that the block still works as expected for MVP.


## Screenshots or screencast

N/A

## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


> Add suggested changelog entry here.
